Networking/Networking/Model/Product/Product.swift

```diff
@@ -290,6 +293,7 @@ public struct Product: Codable {
         let downloadLimit = try container.decode(Int.self, forKey: .downloadLimit)
         let downloadExpiry = try container.decode(Int.self, forKey: .downloadExpiry)
 
+        let buttonText = try container.decodeIfPresent(String.self, forKey: .buttonText) ?? ""
```

Comment:
Do we really want to save it as an empty text? 
If it is not present, it would be better to insert the value as an optional, so as not to save it if not present. What do you think?

Reply:
from my testing and the [WC API example response](https://woocommerce.github.io/woocommerce-rest-api-docs/#retrieve-a-product) (on the right side), `button_text` field is empty instead of `null` if not applicable (this seems to be the default API response format for the other fields as well). That was the reason I created the new `buttonText: String` attribute in https://github.com/woocommerce/woocommerce-ios/pull/2391 as non-optional with default value to an empty string to be consistent with the API response. Also, we don't have to check if the attribute is `nil` or empty in order to know whether it's set this way.

lemme know if you see any concerns with this non-optional attribute!

Reply:
Sounds good to me if this is the reason, but if the field it's always present, why use `decodeIfPresent`? I think that using `decode` is enough.

Reply:
good point! updated in https://github.com/woocommerce/woocommerce-ios/pull/2392/commits/2d33218d2318056543aeaa24132d50e3d9db7e99

I'm always a bit concerned if some site configurations would alter the fields, but I think it's better to make it an exception rather than failing silently.